daisy/prowjob/run/e2e_tests/__main__.py

```diff
@@ -13,40 +13,59 @@
 # limitations under the License.
 
 import argparse
+from Crypto import Random
 import glob
+import multiprocessing
 import os
 import sys
+import tarfile
+import time
 
-from run.call import call
+import apiclient
+from oauth2client.service_account import ServiceAccountCredentials
+
+from run import common
 from run import git
+from run import gcs
 from run import logging
 
 ARGS = []
-PACKAGE_URL = 'github.com/GoogleCloud/compute-image-tools'
-PACKAGE_PATH = os.path.join(os.environ['GOPATH'], 'src', PACKAGE_URL)
+PARALLEL_TESTS = 10
+REPO_OWNER = 'GoogleCloudPlatform'
+REPO_NAME = 'compute-image-tools'
+REPO_URL = 'https://github.com/%s/%s.git' % (REPO_OWNER, REPO_NAME)
+TEST_PROJECT = 'gce-daisy-test'
+TEST_ID = str(common.utc_timestamp())
+TEST_BUCKET = 'gce-daisy-test'
+TESTS_GCS_DIR = os.path.join('tests', TEST_ID)
+WFS_TAR = 'wfs.tar.gz'
+TEST_WFS_GCS = os.path.join(TESTS_GCS_DIR, WFS_TAR)
+
+cloud_builder_client = None
 
 
 def main():
     logging.info('Got --tests=%s', ARGS.tests)
 
-    logging.info('Downloading Daisy')
-    cmd = ['go', 'get', PACKAGE_URL]
-    code = call(cmd).returncode
-    if code:
-        return code
+    logging.info('Fetching Daisy Repo.')
+    repo = git.Repo('repo', clone=REPO_URL)
+    if repo.clone_code:
+        return repo.clone_code
 
-    # commit = 'Something from Argo'
-    repo = git.Repo(PACKAGE_PATH)
-    repo.checkout(branch='master')
+    logging.info('Tar\'ing workflows to upload to GCS.')
+    wf_dir = os.path.join('repo', 'daisy_workflows', ARGS.tests)
+    with tarfile.open('wfs.tar.gz', 'w:gz') as tgz:
+        tgz.add(wf_dir, arcname=os.path.sep)
 
-    wf_dir = os.path.join(PACKAGE_PATH, 'daisy_workflows', ARGS.tests)
-    wfs = glob.glob(os.path.join(wf_dir, '*.wf.json'))
+    logging.info('Uploading tests to %s.', TEST_WFS_GCS)
```

Comment:
are you missing the os.path.join(TEST_BUCKET, TEST_WFS_GCS) ?

Reply:
No, I moved the bucket logic into gcs.py. gcs.upload_file uses the BUCKET in gcs.py.